### ISSUE TITLE
Multi-label action recognition?

### ISSUE THREAD

**Issue by fabienbaradel** [abhinanda-punnakkal/BABEL#1]
Hi,
Thanks for releasing the code and the data, that's super interesting and useful for the community.
I have a question regarding the action recognition experiments.
A high number of action segments contains multiple labels so turning the problem as a simple classification is a bit weird to me.
If I understand well an action segment which has two associated actions will be represented two times in the training set: a first time associated to the first action and then a second time with the other action.. A im right?
The top1 and top5 does not really consider these multi-label classification problem. Did you try training using a binary cross-entropy loss and reporting metrics such as precision, recall, average precision?
Thanks a lot for your explanations

**Comment by zgzxy001** [comment]
Same question here.

**Comment by cotton-ahn** [comment]
I am also confused when looking at this dataset. I need more precise description regarding this problem.

**Comment by carjun** [comment]
Thanks @fabienbaradel and others, for the good question. There are indeed cases where multiple actions overlap (happen simultaneously). 

Task definition: Following previous work, the action recognition task that we define is the following: given a 5-second motion chunk, predict the action in it. @cotton-ahn: For more details about the task definition, please refer to Sec. 5 of the paper. 

For this task, the Top-1 accuracy is a reasonable metric if we assume that a single chunk only contains one action. However, as you note, this is not always the case. 

The Top-5 metric captures the fact that there may be multiple GT actions in a motion. In implementation, the motion span corresponding to each action is considered an individual segment $X_s$, with one GT action. According to the Top-5 metric, the model is successful if the top 5 predicted actions contains the GT action. 

In this setting, the Top-5 metric behaves like a Recall@5 metric. 

We acknowledge that formulating action recognition as a multi-label classification problem with metrics like precision/recall/MAP might be more informative. This was the plan originally, but we couldn't get around to it. If anyone is going forward with implementing a new metric for the Action Recognition task, we are happy to support it.
